### PR TITLE
LPS-175125 Allow attachment file download on the collection display table

### PR DESCRIPTION
**Pull request by carolmariaabb**
Related Issue: https://issues.liferay.com/browse/LPS-175125

### CODE REVIEW

**Comment by carolmariaabb** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/info/item/renderer/object_entry.jsp [hunk 1]
```diff
@@ -34,9 +34,23 @@ for (Map.Entry<String, Serializable> entry : objectEntryValues.entrySet()) {
 				</a>
 			</c:when>
 			<c:otherwise>
-				<span class="text-truncate-inline">
-					<span class="text-truncate"><%= entry.getValue() %></span>
-				</span>
+				<c:choose>
+					<c:when test="<%= entry.getValue() instanceof Link %>">
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/info/item/renderer/object_entry.jsp:38]: Hey @carolmariaabb , I think we also need to cover the scenario where the Object has the Display Page Template (friendlyURL) and there is an attachment type column, this way the other columns will still have the link to redirect to DPT and only the attachment column will have the link to download the file.

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/info/item/renderer/object_entry.jsp (hunk 1 above)
Comment (carolmariaabb) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/info/item/renderer/object_entry.jsp]: @guilhermedcamacho I updated the logic. I'm not sure if I understood correctly, but I made the changes considering what you said:

> other columns will still have the link to redirect to DPT and only the attachment column will have the link to download the file

**Comment by brianchandotcom** [comment]
@carolmariaabb see 4237efee

**Comment by brianchandotcom** [comment]
@carolmariaabb see 4237efee in my origin. I merged this PR.

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/info/item/renderer/ObjectEntryRowInfoItemRenderer.java [hunk 2]
```diff
@@ -190,7 +212,39 @@ else if (Objects.equals(
...
+							{
+								href = StringBundler.concat(
+									_portal.getPathContext(),
+									_portal.getPathMain(), "/portal/login");
+								label = dlFileEntry.getFileName();
+							}
+						};
+
+						try {
+							FileEntry fileEntry = _dlAppService.getFileEntry(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/info/item/renderer/ObjectEntryRowInfoItemRenderer.java]: Hey @carolmariaabb , I don't know what's the difference between DLFileEntry and FileEntry, but I think we can get rid of the use of method fetchDLFileEntry invoked before and just use FileEntry to get the file name like I saw [here](https://github.com/liferay/liferay-portal/blob/b38777bdf00d43dcb9d978655717a3c5886179e1/modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/java/com/liferay/object/dynamic/data/mapping/form/field/type/internal/attachment/AttachmentDDMFormFieldTemplateContextContributor.java#L151). Is that possible?

Comment (guilhermedcamacho) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/info/item/renderer/ObjectEntryRowInfoItemRenderer.java]: I found this [documentation](https://help.liferay.com/hc/en-us/articles/360029045591-Getting-Files) and after reading it, it seemed to me that the correct entity to use is FileEntry. Let me know if you have more context about it.

Comment (carolmariaabb) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/info/item/renderer/ObjectEntryRowInfoItemRenderer.java]: I created this [Technical Debt](https://issues.liferay.com/browse/LPS-175512) to address this when possible

**Pull request closed without merging**